etc/setup/scripts/subtasks/keycloak_setup.sh

```diff
@@ -0,0 +1,226 @@
+    "${KEYCLOAK_SERVICE_PUBLIC_URL}/auth/admin/realms/${realm}/clients" | jq -S .
+}
+
+
+#################################
+set_client () {
+  realm=$1
+  client=$2
+  listen=$3
+  redirect=$4
+
+  # Will add / to listen only if it is present
+
+  JSON='{
+    "clientId": "'"${client}"'",
+    "enabled": true,
+    "protocol": "openid-connect",
+    "implicitFlowEnabled": true,
+    "standardFlowEnabled": true,
+    "publicClient": false,
+    "redirectUris": [
+      "'${CANDIG_PUBLIC_URL}${redirect}'"
+    ],
+    "attributes": {
+      "saml.assertion.signature": "false",
+      "saml.authnstatement": "false",
+      "saml.client.signature": "false",
+      "saml.encrypt": "false",
+      "saml.force.post.binding": "false",
+      "saml.multivalued.roles": "false",
+      "saml.onetimeuse.condition": "false",
+      "saml.server.signature": "false",
+      "saml.server.signature.keyinfo.ext": "false",
+      "saml_force_name_id_format": "false"
+    }
+  }'
+  echo $JSON
+  curl \
+    -H "Authorization: bearer ${KC_TOKEN}" \
+    -X POST -H "Content-Type: application/json"  -d "${JSON}" \
+    "${KEYCLOAK_SERVICE_PUBLIC_URL}/auth/admin/realms/${realm}/clients"
+}
+
+get_secret () {
+  id=$(curl -H "Authorization: bearer ${KC_TOKEN}" \
+    ${KEYCLOAK_SERVICE_PUBLIC_URL}/auth/admin/realms/${KC_REALM}/clients 2> /dev/null \
+    | python3 -c 'import json,sys;obj=json.load(sys.stdin); print([l["id"] for l in obj if l["clientId"] ==
+    "'"$KC_CLIENT_ID"'" ][0])')
+
+  curl -H "Authorization: bearer ${KC_TOKEN}" \
+    ${KEYCLOAK_SERVICE_PUBLIC_URL}/auth/admin/realms/${KC_REALM}/clients/$id/client-secret 2> /dev/null |\
+    python3 -c 'import json,sys;obj=json.load(sys.stdin); print(obj["value"])'
+}
+##################################
+
+# SCRIPT START
+
+echo "-- Starting setup calls to keycloak --"
+echo "$KC_ADMIN_USER $KC_ADMIN_PW $KEYCLOAK_SERVICE_PUBLIC_URL"
+
+echo ">> Getting KC_TOKEN .."
+KC_TOKEN=$(get_token)
+#echo ">> retrieved KC_TOKEN ${KC_TOKEN}"
+echo ">> .. got it..."
+
+echo ">> Creating Realm ${KC_REALM} .."
+set_realm ${KC_REALM}
+echo ">> .. created..."
+
+
+echo ">> Setting client KC_CLIENT_ID .."
+set_client ${KC_REALM} ${KC_CLIENT_ID} "${TYK_LISTEN_PATH}" ${KC_LOGIN_REDIRECT_PATH}
+echo ">> .. set..."
+
+echo ">> Getting KC_SECRET .."
+# TODO: Fix;
+# WARNING: Despite being exported,
+# KC_SECRET is not properly useable
+# elsewhere in the setup process.
+# e.g.: Tyk (needs manual intervention)
+export KC_SECRET=$(get_secret  ${KC_REALM})
+echo "** Retrieved KC_SECRET as ${KC_SECRET} **"
+echo ">> .. got it..."
+
+echo ">> Adding user .."
+add_user
+echo ">> .. added..."
+
+echo 
+echo ">> .. waiting for keycloak to restart..."
+while !  docker logs --tail 5  ${CONTAINER_NAME_CANDIG_AUTH} | grep "Admin console listening on http://127.0.0.1:9990" ; do sleep 1 ; done
+docker exec ${CONTAINER_NAME_CANDIG_AUTH}  rm /opt/jboss/keycloak/standalone/configuration/keycloak-add-user.json 2> /dev/null
+echo ">> .. ready..."
```

Comment:
looks fine, but would recommend moving scripts specific to a service in lib/ with the service